packages/backend/src/apps/custom-api/actions/http-request/index.ts

```diff
@@ -59,14 +61,22 @@ export default defineAction({
       throw new Error('The URL you are trying to call is not allowed.')
     }
 
-    const response = await $.http.request({
-      url,
-      method,
-      data,
-      // Redirects open up way too many vulns (e.g. someone changes the
-      // redirect target to a malicious endpoint), so disable it.
-      maxRedirects: 0,
-    })
+    const response = await $.http
+      .request({
+        url,
+        method,
+        data,
+        // Redirects open up way too many vulns (e.g. someone changes the
+        // redirect target to a malicious endpoint), so disable it.
+        maxRedirects: 0,
+      })
+      .catch((err) => {
+        throw new StepError({
+          name: 'Custom API error',
+          solution: 'Check your custom app and retry again',
+          httpErrorDetails: customiseHttpErrorDetails(err),
```

Comment:
Curious why we need this instead of integrating http status codes, etc into

_Rationale_
I'm worried this is a bit too specialized. What happens if we end up supporting other transports like SOAP, MQTT etc, will we need to create `soapErrorDetails` / `mqttErrorDetails`?

(I could also be thinking a bit too much. But I generally lean towards avoiding edge cases when in doubt.)